File: src/main/java/sx/event/EventDelegate.java
package sx.event;

/**
 * Multicast event delegate interface
 * Created by masc on 11.08.14.
 */
public interface EventDelegate<T extends EventListener> {
    /** Add listener to this multicast delegate */
    void add(T listener);
    /** Remove listener from this multicast delegate */
    void remove(T listener);
}

File: src/main/java/sx/event/EventDelegate.kt
package sx.event

/**
 * Multicast event delegate interface
 * Created by masc on 11.08.14.
 */
interface EventDelegate<T : EventListener> {
    /** Add listener to this multicast delegate  */
    fun add(listener: T)
    /** Remove listener from this multicast delegate  */
    fun remove(listener: T)
}
